contracts/cosmwasm-vm/cw-ibc-core/src/ics04_channel/packet/receive_packet.rs

```diff
@@ -66,9 +63,22 @@ impl<'a> CwIbcCoreContext<'a> {
             })
             .map_err(Into::<ContractError>::into)?;
         }
+
+        if channel_end.order_matches(&Order::Ordered) {
+            let next_seq_recv =
+                self.get_next_sequence_recv(deps.storage, &dst_port, &dst_channel)?;
+            if packet_sequence > next_seq_recv {
+                return Err(PacketError::InvalidPacketSequence {
+                    given_sequence: packet_sequence,
+                    next_sequence: next_seq_recv,
+                })
+                .map_err(Into::<ContractError>::into)?;
+            }
+        }
+
```

Comment:
Is this not duplicate logic from lines 148-163

Reply:
not exactly duplicate one is checking if already received while other is validating packet sequence. But can be merged in a single function . Let me update that.